src/GlobalTarget.js

```diff
@@ -33,24 +33,18 @@ class GlobalTarget extends Component {
 
   componentDidMount() {
     this.renderInRemote = ({ jsx, context, contextTypes, id }) => {
-      const newStackItem = {
-        jsx,
-        SetContextComponent: createSetContextComponent(contextTypes),
-        context,
-        id
-      };
-      const stackTypes = this.state.stack.map(({jsx}) => jsx.type);
-      if (stackTypes.filter(type => type === jsx.type).length === 0) {
-
+      const stackItem = this.state.stack.filter(({id: stackId}) => stackId === id)[0];
```

Comment:
Should we be more careful here and check first the length of `stack` before attempting to extract first item?

Reply:
not really. stack can never be undefined.

[].filter(callback)[0] will result in undefined. This check is done just below it.